tests/unit/api/test_api.py

```diff
@@ -147,57 +147,73 @@ def test_initialize_options(self):
-            Set env_name environment variable to env_value
-            Assert api.attr_name = env_value
-            """
-            os.environ[env_name] = env_value
-            initialize()
-            self.assertEqual(getattr(api, attr_name), env_value)
-
-        @preserve_environ_datadog
-        def test_api_params_default(env_name, attr_name, expected_value):
-            """
-            Unset env_name environment variable
-            Assert api.attr_name = expected_value
-            """
-            if os.environ.get(env_name):
-                del os.environ[env_name]
-            initialize()
-            self.assertEqual(getattr(api, attr_name), expected_value)
-
-        @preserve_environ_datadog
-        def test_api_params_from_params(env_name, parameter, attr_name, value):
-            """
-            Unset env_name environment variable
-            Initialize API with parameter=value
-            Assert api.attr_name = value
-            """
-            if os.environ.get(env_name):
-                del os.environ[env_name]
-            initialize(api_host='http://localhost')
-            self.assertEqual(api._api_host, 'http://localhost')
-
-        # Default values
-        test_api_params_default("DATADOG_API_KEY", "_api_key", None)
-        test_api_params_default("DATADOG_APP_KEY", "_application_key", None)
-        test_api_params_default("DATADOG_HOST", "_api_host", "https://api.datadoghq.com")
-
-        # From environment
-        test_api_params_from_env("DATADOG_API_KEY", "_api_key", env_value="apikey")
-        test_api_params_from_env("DATADOG_APP_KEY", "_application_key", env_value="appkey")
-        test_api_params_from_env("DATADOG_HOST", "_api_host", env_value="http://localhost")
-
-        # From parameters
-        test_api_params_from_params("DATADOG_API_KEY", "api_key", "_api_key", "apikey2")
-        test_api_params_from_params("DATADOG_APP_KEY", "app_key", "_application_key", "appkey2")
-        test_api_params_from_params("DATADOG_HOST", "api_host", "_api_host", "http://127.0.0.1")
+    def test_default_values(self):
+        initialize()
+
+        self.assertIsNone(api._api_key)
+        self.assertIsNone(api._application_key)
+        self.assertEqual(api._api_host, "https://api.datadoghq.com")
+        self.assertEqual(api._host_name, util.hostname.get_hostname())
+
+    def test_env_var_values(self):
+        os.environ["DATADOG_API_KEY"] = "API_KEY_ENV"
+        os.environ["DATADOG_APP_KEY"] = "APP_KEY_ENV"
+        os.environ["DATADOG_HOST"] = "HOST_ENV"
+
+        initialize()
+
+        self.assertEqual(api._api_key, "API_KEY_ENV")
+        self.assertEqual(api._application_key, "APP_KEY_ENV")
+        self.assertEqual(api._api_host, "HOST_ENV")
+        self.assertEqual(api._host_name, util.hostname.get_hostname())
+
+        del os.environ["DATADOG_API_KEY"]
+        del os.environ["DATADOG_APP_KEY"]
+        del os.environ["DATADOG_HOST"]
+
+    def test_function_param_value(self):
```

Comment:
Should we add a quick test that tests the order these are used? Like provided attribute >> existing attribute >> environment variable?